storage/storage_image.go

```diff
@@ -137,10 +137,9 @@ func (s *storageImageDestination) putBlob(stream io.Reader, blobinfo types.BlobI
 		Digest: "",
 		Size:   -1,
 	}
-	// Try to read an initial snippet of the blob.
-	header := make([]byte, 10240)
-	n, err := stream.Read(header)
-	if err != nil && err != io.EOF {
+	buf := [archive.HeaderSize]byte{}
+	n, err := io.ReadAtLeast(stream, buf[:], len(buf))
+	if err != nil && err != io.EOF && err != io.ErrUnexpectedEOF {
```

Comment:
I'm not feeling warm and fuzzy including ErrUnexpectedEOF in this if statement.  If we get an error in the middle of reading a fixed-size block or data structure, I think we ought to error out.  My $.02.  Take it with a grain of salt that I'm still coming up to speed in golang.